junit-jupiter-engine/src/test/java/org/junit/jupiter/engine/discovery/DiscoverySelectorResolverTests.java

```diff
@@ -287,16 +309,31 @@ void methodResolutionByUniqueIdWithParams() {
 		List<UniqueId> uniqueIds = uniqueIds();
 		assertThat(uniqueIds).contains(uniqueIdForClass(HerTestClass.class));
 		assertThat(uniqueIds).contains(uniqueIdForMethod(HerTestClass.class, "test7(java.lang.String)"));
+		// @formatter:off
+		assertThat(listener.stream()
+				.filter(logRecord -> JavaElementsResolver.class.getName().equals(logRecord.getLoggerName()))
+		).isEmpty();
+		// @formatter:on
 	}
 
 	@Test
-	void resolvingUniqueIdWithWrongParamsResolvesNothing() {
-		UniqueIdSelector selector = selectUniqueId(
-			uniqueIdForMethod(HerTestClass.class, "test7(java.math.BigDecimal)").toString());
-		EngineDiscoveryRequest request = request().selectors(selector).build();
+	@TrackLogRecords
+	void resolvingUniqueIdWithWrongParamsResolvesNothing(LogRecordListener listener) {
+		UniqueId uniqueId = uniqueIdForMethod(HerTestClass.class, "test7(java.math.BigDecimal)");
+		EngineDiscoveryRequest request = request().selectors(selectUniqueId(uniqueId)).build();
 
 		resolver.resolveSelectors(request, engineDescriptor);
 		assertTrue(engineDescriptor.getDescendants().isEmpty());
+		// @formatter:off
+		assertThat(listener.stream()
+				.filter(logRecord -> logRecord.getLevel() == Level.WARNING)
+				.filter(logRecord -> JavaElementsResolver.class.getName().equals(logRecord.getLoggerName()))
+				.map(LogRecord::getMessage)
```

Comment:
You should be able to remove the two filters by using `org.junit.platform.commons.logging.LogRecordListener.stream(Class<?>, Level)`.

Reply:
Very convenient 👍

Reply:
glad you like it!